### PR TITLE
Feature: subscript JSON using a typed key path

### PR DESCRIPTION
**Pull request by cjmconie**
This PR explores a method of subscripting JSON types using a typed key path.

### CODE REVIEW

**Review by zoul — COMMENTED** [review]
I was initially in favour of this instead of #15 (subscripting using a plain string key path), but in the end, after reading the code, what does the extra `KeyPath` type really offer? There’s no additional type safety to speak of, but we would have to expose (and then maintain) a new type in the API. And the type would really have no extra use besides being needed as an argument to `subscript(keyPath:)`, so it doesn’t sound like a good deal. Do you have a preference?

File: GenericJSONTests/QueryingTests.swift [hunk 1]
```diff
@@ -58,4 +58,27 @@ class QueryingTests: XCTestCase {
...
+                "str": "col",
+                "arr": [1, 2, 3],
+                "obj": [
+                    "x": "rah",
+                    "y": "tar",
+                    "z": "yaz"
+                ]
+            ]
+        ]
+        XCTAssertEqual(json[keyPath: KeyPath("string")], JSON.string("foo bar"))
```
Comment (zoul) [inline GenericJSONTests/QueryingTests.swift:77]: By the way, it’s possible to leave out the `KeyPath` type name here:

    XCTAssertEqual(json[keyPath: "string"], JSON.string("foo bar"))

This is the whole point of the `ExpressibleByStringLiteral` conformance.

Comment (cjmconie) [reply GenericJSONTests/QueryingTests.swift:77]: Thanks 😃

**Comment by cjmconie** [comment]
I agree. `KeyPath` adds little benefit in this scenario. A vanilla string key path is simpler to implement and maintain.

I am happy for this PR to be closed.

**Comment by zoul** [comment]
We decided to do #15 instead.

**Pull request closed without merging**